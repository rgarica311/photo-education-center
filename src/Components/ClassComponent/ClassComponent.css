.classPhoto {
  height: 40vh;
  width: 100vw;
  position: relative;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  display: flex;
  align-items: center;
  justify-content: center;
  font-family: Raleway;
  font-weight: 900;
  font-size: 4rem;
  color: white;
}

.photo {
  background-image: url('https://s4c5f7q2.stackpathcdn.com/clevercourse/wp-content/uploads/2014/08/photodune-6243139-vintage-photography-m.jpg');
  height: 490px;
  width: 690px;
  background-position: center;
  background-size: cover;
  margin-right: 50px;
}


.dateColumn {
  display: flex;
  flex-direction: column;
}

.largeBoldSpan{
  font-weight: bold;
  font-size: 1.5rem;
}


.sReviews {
  margin-top: 10px;
}

.flexRow {
  display: flex;
  margin-bottom: 10px;
}

.flexColumn {
  display: flex;
  flex-direction: column;
  padding-left: 50px;
  padding-right: 50px;
}

li {
  color: black;
}

.detsilUlContainer {
  width: 100%;
}

@media screen and (max-width: 1366px) {
  .classPhoto {
    height: 15vh;
  }

  .photo {
    height: 20vh;
    width: 100%;
    margin-right: 0;
    margin-bottom: 20px;
  }

  div.datesDetailsRow {
    padding-left: 10px;
    padding-right: 10px;
    display: flex;
    flex-direction: column;
  }

  ul.detailsContainerCol {
    width: 100%;
  }

  div.detailUlContainer {
    width: 100%;
    height: 80vh;
  }

  p.beginnerClassDetail {
    position: relative;
    top: 0;
    margin-bottom: 20px;
    padding: 25px;
    font-family: kadwa;
    text-align: center;
    margin: 0 auto;
  }

  



}
