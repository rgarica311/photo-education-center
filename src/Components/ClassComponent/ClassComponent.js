import React, { Component } from 'react';
import './ClassComponent.css';
import Detail from '../Detail/Detail';
import CpaContext from '../../cpaContext'
export default class ClassComponent extends Component {
  constructor(props){
    super(props);
    this.state = {
      classDesc: props.classDesc,
      photoUrl: this.props.photoUrl,
      title: this.props.title
    }
  }
  render() {
    console.log('this.state.classDesc class component', this.state.classDesc)
    return (
      <CpaContext.Consumer>
        {(context) => (
          <div className='classDetail'>
          {this.state.photoUrl
            ? <div style={{backgroundImage: `url(${this.state.photoUrl})`}} className='classPhoto bannerPhoto'>{this.state.title}</div>
            : null }

            <div className='datesDetailsRow'>
              <div className='photo'></div>
              <p className='beginnerClassDetail'>{this.state.classDesc}</p>
              {this.state.classDesc !== undefined ? <Detail title={this.state.title} classDesc={this.state.classDesc} /> : null}
            </div>
          </div>
        )}
    </CpaContext.Consumer>
    )
  }
}
