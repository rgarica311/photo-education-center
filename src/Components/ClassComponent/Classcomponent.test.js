import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ClassComponent from './ClassComponent';

describe(`ClassComponent Component`, () => {

  it('renders ClassComponent by default', () => {
    const wrapper = shallow(<ClassComponent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
