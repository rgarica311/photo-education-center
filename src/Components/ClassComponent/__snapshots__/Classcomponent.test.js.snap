// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassComponent Component renders ClassComponent by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
